Sources/ArgumentParser/Parsable Properties/Argument.swift

```diff
@@ -83,6 +83,42 @@ extension Argument where Value: ExpressibleByArgument {
   }
 }
 
+/// The strategy to use when parsing multiple values from `@Option` arguments
+/// into an array.
+public enum ArgumentArrayParsingStrategy {
+  /// Parse only unprefixed values from the command-line input, ignoring
+  /// any inputs that have a dash prefix.
+  ///
+  /// For example, for a parsable type defined as following:
+  ///
+  ///     struct Options: ParsableArguments {
+  ///         @Flag() var verbose: Bool
+  ///         @Argument(parsing: .remaining) var words: [String]
+  ///     }
+  ///
+  /// Parsing the input `--verbose one two` or `one two --verbose` would result
+  /// in `Options(verbose: true, words: ["one", "two"])`. Parsing the input
+  /// `one two --other` would result in an unknown option error for `--other`.
+  ///
+  /// This is the default strategy for parsing argument arrays.
+  case remaining
+  
+  /// Parse all remaining inputs after parsing any known options or flags,
+  /// including dash-prefixed inputs.
+  ///
+  /// For example, for a parsable type defined as following:
+  ///
+  ///     struct Options: ParsableArguments {
+  ///         @Flag() var verbose: Bool
+  ///         @Argument(parsing: .remainingValues) var words: [String]
+  ///     }
+  ///
+  /// Parsing the input `--verbose one two --other` would include the `--other`
+  /// flag in `words`, resulting in
+  /// `Options(verbose: true, words: ["one", "two", "--other"])`.
+  case unconditionalRemaining
```

Comment:
`ArrayParsingStrategy.remaining` has some guidance as to how this usually should be done (i.e. using a `--` terminator on the input). We should probably add a link here to that discussion and make sure they line up. And generally give some guidance as to when `.remaining` vs. `. unconditionalRemaining` is a good choice for developing new tools. I’d argue that `.remaining` is the _correct_ choice and have users of the tool use the `--` escape / termination.

Reply:
I added a note here to give a bit more context. When we have documentation of the `--` behavior, let's add a link from here as well.